src/coreclr/jit/emitarm64.h

```diff
@@ -294,6 +295,24 @@ static emitter::floatImm8 emitEncodeFloatImm8(double immDbl);
 
 static double emitDecodeFloatImm8(const emitter::floatImm8 fpImm);
 
+static ssize_t emitEncodeRotationImm90_or_270(ssize_t imm);
```

Comment:
it just occurred to me that why can't we just have `IsValid*` methods that checks for `imm == 90` or `imm == 180`, etc. and then during encoding just `/= 90` or `*= 90`? Something along the lines of https://github.com/dotnet/runtime/pull/98187/commits/f75c3ced91ddcfad6653062e71a1c0f5548c2611#diff-2b2c8b9011607926410624d6f81613fad7b74c6e0516d578675a8b792998fe4fR12094-R12096

Reply:
FYI, #98187 adds `isValidRot`, which checks if the immediate is a multiple of 90, between 0 and 270.

Reply:
Except for the formats that @TIHan is adding, you will need to have methods like `isValid0_or_270()` because they don't take 90 or 180.

Reply:
Ah I see. In my next batch, I have an encoding where the rotation can only be 90 or 270. If you'd like, I can add methods like `isValid0_or_270`, `isValid90_or_270`, etc to that PR.

Reply:
> it just occurred to me that why can't we just have IsValid* methods that checks for imm == 90 or imm == 180, etc. and then during encoding just /= 90 or *= 90?

The `emitEncodeRotationImm*` functions will assert the values do not match the cases. Personally, I wouldn't do `/=` 90 or `*= 90` as it just feels clever. It's clearer to see the mapping of `0 => 90` and `1 => 270`.

Reply:
> The emitEncodeRotationImm* functions will assert if the values do not match the cases

Hhm, won't the `IsValid*` method do that when you call appropriate method like `IsValidRot()` vs. `IsValidRot_0_or_270`?

Reply:
If @amanasifkhalid wants to use them he can since he has them in his PR. It doesn't matter either way. At this point we are bike shedding.

Reply:
One of the argument that goes in favor of `emit*codeRotationImm()` is sometimes `90` would be encoded as `0` and other times it will be `1`, so @amanasifkhalid - something to keep in mind.

```c++
/*static*/ ssize_t emitter::emitEncodeRotationImm90_or_270(ssize_t imm)
{
    switch (imm)
    {
        case 90:
            return 0;
 ....
}

/*static*/ ssize_t emitter::emitEncodeRotationImm0_to_270(ssize_t imm)
{
    switch (imm)
    {
        case 0:
            return 0;

        case 90:
            return 1;
  ...
}
```